### PR TITLE
Change warning and exception logger for retries

### PR DESCRIPTION
**Pull request by jzbahrai**
# Summary | Résumé

We don't want the API to throw an error for one retry. We will throw a warning when there have been more than 10 retries.

This is from where I assume we have retry data:
https://github.com/celery/celery/blob/main/celery/app/task.py#L704

### CODE REVIEW

File: app/celery/provider_tasks.py [hunk 1]
```diff
@@ -76,7 +76,10 @@ def deliver_email(self, notification_id):
     except Exception as e:
         try:
             current_app.logger.warning(f"The exception is {repr(e)}")
-            current_app.logger.exception("RETRY: Email notification {} failed".format(notification_id))
+            if self.request.retries <= 10:
+                current_app.logger.warning("RETRY {}: Email notification {} failed".format(self.request.retries, notification_id))
+            else:
+                current_app.logger.exception("RETRY: Email notification {} failed".format(notification_id))
```
Comment (sastels) [inline app/celery/provider_tasks.py:82]: yeah, this is good. I was thinking that we could just let the `MaxRetriesExceededError` cause an error, but we set `max_retries` to 48, and maybe we want to see an error before then 👍

Comment (sastels) [reply app/celery/provider_tasks.py:82]: I was thinking that we should add some tests to make sure this code doesn't cause a problem, but we already have a test around "can't find the notification in the database, just retry it" that will run this new code.
https://github.com/cds-snc/notification-api/blob/main/tests/app/celery/test_provider_tasks.py#L80-L90

**Review by sastels — APPROVED** [review]
:shipit:

File: app/celery/provider_tasks.py (hunk 1 above)
Comment (jzbahrai) [reply app/celery/provider_tasks.py:82]: thanks! ill merge

**Pull request merged**